src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -1250,3 +1255,53 @@ impl CatalogState {
         row
     }
 }
+
+mod name_simplifier {
+    use std::str::FromStr;
+
+    use mz_repr::GlobalId;
+    use mz_sql::ast::visit_mut::VisitMut;
+    use mz_sql::ast::{Raw, RawClusterName, RawItemName};
+    use mz_sql::names::{PartialItemName, SchemaSpecifier};
+    use mz_sql::normalize;
+
+    // Used when displaying a view's source for human creation. If the name
+    // specified is the same as the name in the catalog, we don't use the ID format.
+    #[derive(Debug)]
+    pub struct NameSimplifier<'a> {
+        pub catalog: &'a super::CatalogState,
+    }
+
+    impl<'ast, 'a> VisitMut<'ast, Raw> for NameSimplifier<'a> {
+        fn visit_cluster_name_mut(&mut self, node: &mut RawClusterName) {
+            if let RawClusterName::Resolved(name) = node {
+                if self.catalog.clusters_by_name.contains_key(name) {
+                    *node = RawClusterName::Unresolved(name.to_string().into());
+                }
+            }
+        }
+
+        fn visit_item_name_mut(&mut self, node: &mut RawItemName) {
+            if let RawItemName::Id(id, name) = node {
```

Comment:
Is this block covered by a test? Looks interesting to test

Reply:
This was re-added by merge skew.